### PR TITLE
Add console app framework from helix and scaffold for secret management cli

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/Microsoft.DncEng.SecretManager/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "ApplicationInsightsInstrumentationKey": "00000000-0000-0000-0000-000000000000"
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.SecretManager/appsettings.json:2]: ? zero?

File: src/Shared/Microsoft.DncEng.CommandLineLib/Authentication/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.DncEng.CommandLineLib.Authentication
+{
+    public static class Constants
```
Comment (ChadNedzlek) [inline src/Shared/Microsoft.DncEng.CommandLineLib/Authentication/Constants.cs:7]: Generic "constants" classes always get a little weird.

It seems like right now the only class that actually uses the tenantId is the AzureServiceTokenProviderCredential class. Could it be a member of that class?

File: src/Microsoft.DncEng.SecretManager/Program.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+using Microsoft.DncEng.CommandLineLib.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DncEng.SecretManager
+{
+    public class Program : DependencyInjectedConsoleApp
+    {
+        static async Task<int> Main(string[] args)
+        {
+            return await new Program().RunAsync(args);
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.SecretManager/Program.cs:18]: Does it need the "async" "return await"? Could it just return the Task itself?

File: src/Microsoft.DncEng.SecretManager/Program.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+using Azure.Core;
+using Microsoft.DncEng.CommandLineLib;
+using Microsoft.DncEng.CommandLineLib.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DncEng.SecretManager
+{
+    public class Program : DependencyInjectedConsoleApp
+    {
+        static async Task<int> Main(string[] args)
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.SecretManager/Program.cs:16]: I'd personally include the access modifier here. I can never remember the rules about what things have what defaults, so it's easier to just write it down.

**Review by ChadNedzlek — APPROVED** [review]

File: src/Microsoft.DncEng.SecretManager/appsettings.json (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/appsettings.json:2]: We haven't created an app insights for this yet. I will add the key here once we have one, but I wanted to get the projects in so we can start work on implementations.

File: src/Shared/Microsoft.DncEng.CommandLineLib/Authentication/Constants.cs (hunk 2 above)
Comment (alexperovich) [reply src/Shared/Microsoft.DncEng.CommandLineLib/Authentication/Constants.cs:7]: I will move it.

File: src/Microsoft.DncEng.SecretManager/Program.cs (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Program.cs:18]: Yep, I will do that.

**Review by chcosta — APPROVED** [review]
Just the one comment (which we've discussed in Team's).  Everything else looks sane.  Thanks for getting this going!

**Comment by alexperovich** [comment]
We have decided that using Mono.Options is acceptable for now. It doesn't add any debt and the Cli epic can easily switch things once it gets funded. Merging.

**Pull request merged**